runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPlugin.java

```diff
@@ -152,6 +157,23 @@ public void onChatMessage(ChatMessage event)
 	@Subscribe
 	void onTick(GameTick tick)
 	{
+		boolean inRegion = client.getLocalPlayer().getWorldLocation().getRegionID() == REGION;
```

Comment:
This should probably be done in some region changed hook instead of game tick.

Reply:
Well it would not, but I dont like sticking everything thi GameTick because why not

Reply:
To check region change you should check for gamestate changed to logged in.